pkg/wireguard/wireguard.go

```diff
@@ -380,9 +380,9 @@ func (dev *Device) handlePeerEvent(logger *zap.Logger, peerEvents []PeerEvent) e
 				handleErr = handler.HandlePeerAdded(peerEvent)
 			}
 
-			peerEvents = slices.Delete(peerEvents, i, i+1)
-
 			if handleErr != nil {
+				peerEvents = slices.Delete(peerEvents, i, i+1)
```

Comment:
not sure why we need to delete them? we just range over all peer events?

Reply:
it's also unclear about the logic below on line 391

Reply:
if we delete while in the loop, we're certainly going to miss some events either way

Reply:
so I think the logic was to skip the peers which failed in the handler, but this should be some other way, e.g. by setting the event to some invalid state, so that we skip it later on.

Reply:
Yeah, I think that would be better